### PR TITLE
[GSoC] Move external libraries import inside functions

### LINKED ISSUE #19930: sympy.stats importing Theano automatically. It should not happen
```python
In [2]: from sympy.stats import *
WARNING (theano.configdefaults): g++ not available, if using conda: `conda install m2w64-toolchain`
c:\users\francescobonazzi\miniconda3\lib\site-packages\theano\configdefaults.py:560: UserWarning: DeprecationWarning: there is no c++ compiler.This is deprecated and with Theano 0.11 a c++ compiler will be mandatory
  warnings.warn("DeprecationWarning: there is no c++ compiler."
WARNING (theano.configdefaults): g++ not detected ! Theano will be unable to execute optimized C-implementations (for both CPU and GPU) and will default to Python implementations. Performance will be severely degraded. To remove this warning, set Theano flags cxx to an empty string.
WARNING (theano.tensor.blas): Using NumPy C-API based implementation for BLAS functions.
```

This warning should not happen. NumPy / Theano / PyMC3 backends should only be called inside a method using them. They should not be imported when importing `sympy.stats`.

### PR DESCRIPTION
**Pull request by Smit-create**
#### References to other Issues or PRs

FIxes #19930 

#### Release Notes

NO ENTRY

ping @Upabjojr @czgdp1807

### CODE REVIEW

File: sympy/stats/crv.py [hunk 1]
```diff
@@ -174,76 +171,78 @@ class SampleContinuousNumpy:
...
+                low=float(dist.left), high=float(dist.right), size=size)
+        }
+
+        dist_list = numpy_rv_map.keys()
 
         if dist.__class__.__name__ not in dist_list:
             return None
 
-        return cls.numpy_rv_map[dist.__class__.__name__](dist, size)
+        return numpy_rv_map[dist.__class__.__name__](dist, size)
```
Comment (eric-wieser) [inline sympy/stats/crv.py:203]: While you're here, this simplifies to
```python
return numpy_rv_map.get(dist.__class__.__name__, lambda dist, size: None)(dist, size)
```
or if that feels ugly
```python
f = numpy_rv_map.get(dist.__class__.__name__)
if f is None:
    return None
return f(dist, size)
```

Comment (eric-wieser) [reply sympy/stats/crv.py:203]: Although I suppose there are a lot of these, so fine to leave to another PR.

**Comment by czgdp1807** [comment]
Not a big change. LGTM. +1 for merge.

**Comment by Smit-create** [comment]
@Upabjojr Can this be merged now?

File: sympy/stats/crv.py (hunk 1 above)
Comment (Upabjojr) [reply sympy/stats/crv.py:203]: I would leave this for another PR. We should use subclasses rather than dictionaries.

**Review by eric-wieser — APPROVED** [review]
Preventing the imports certainly seems useful, and the cost repeated dictionary construction is probably cheap compared to the sampling time.

Diff is obviously fine, after hiding whitespace changes.

**Comment by eric-wieser** [comment]
Might be good to squash the commits.

**Comment by asmeurer** [comment]
I guess we should update the `bin/test_external_imports` script to also test external imports for optional submodules.

**Pull request merged**